### PR TITLE
Add jest tests

### PR DESCRIPTION
**Pull request by doteq**

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = {
+  preset: 'ts-jest',
```
Comment (NPM8) [inline jest.config.js]: I think that we can create tests in typescript to maintain type consistency in some of our returning function

File: src/__tests__/login.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const wulkanowy = require('../index');
+const client = new wulkanowy.Client('fakelog.cf');
+
+test('Login to fakelog account', () => {
```
Comment (NPM8) [inline src/__tests__/login.js]: I think that writing it in look like 
```javascript
description('Client', () => {
  description('login method', () => {
    it('login to fakelog accont', () => {...});
  });
});
```

**Comment by NPM8** [comment]
Nah better convention is with `__test__` folder and e2e tests :-D in separated folder because most paths will go deep into tree and it's sometimes confusing in TS or JS if u are not using any kind of module binding :-D e2e tests can be seated up in the main folder because they mostly use default export only so path want is as much confusioning.

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+  testEnvironment: 'node',
+  testRegex: ["(/__tests__/.*|(\\.|/)(test|spec))\\.[jt]sx?$",
+  "(/tests/.*|(\\.|/)(test|spec))\\.[jt]sx?$"]
```
Comment (NPM8) [inline jest.config.js]: Change extension to `ts` we want only test ts files, not js ones.

Comment (NPM8) [inline jest.config.js]: `test|spec` change to `e2e|unit`

File: package.json [hunk 4]
```diff
@@ -6,9 +6,9 @@
   "types": "dist/index.d.ts",
   "scripts": {
     "build": "tsc",
-    "lint": "./node_modules/.bin/eslint src/index.ts",
-    "lint:fix": "./node_modules/.bin/eslint src/index.ts --fix",
-    "test": "npm run lint"
+    "lint": "eslint src/**.ts",
```
Comment (NPM8) [inline package.json]: It won't work properly try `eslint -c .eslintrc.json -ext .ts src/** test/**`. Test it by making an error in any test and source file.

File: tests/client.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+const wulkanowy = require('../src/index.ts');
```
Comment (dominik-korsa) [inline tests/client.e2e.ts]:
```suggestion
import * as wulkanowy from '../src/index';
```

Comment (NPM8) [reply tests/client.e2e.ts]: Even only `../src` should done the work.

**Pull request merged**